src/main/kotlin/dev/supergrecko/kllvm/types/Type.kt

```diff
@@ -1,15 +1,16 @@
-package dev.supergrecko.kllvm.core.typedefs
+package dev.supergrecko.kllvm.types
 
-import dev.supergrecko.kllvm.core.enumerations.TypeKind
+import dev.supergrecko.kllvm.contracts.ContainsReference
 import dev.supergrecko.kllvm.core.message.Message
-import dev.supergrecko.kllvm.core.types.*
+import dev.supergrecko.kllvm.core.typedefs.Context
+import dev.supergrecko.kllvm.core.typedefs.Value
 import dev.supergrecko.kllvm.core.values.IntValue
 import dev.supergrecko.kllvm.utils.toBoolean
 import org.bytedeco.llvm.LLVM.LLVMTypeRef
 import org.bytedeco.llvm.global.LLVM
 
-public open class Type internal constructor() {
-    internal lateinit var ref: LLVMTypeRef
+public open class Type internal constructor() : ContainsReference<LLVMTypeRef> {
+    public final override lateinit var ref: LLVMTypeRef
```

Comment:
The downside to the contract solution is that the ref has to be public for this to work as we can't declare internal properties in interfaces